Frends.AmazonS3.DownloadObject/Frends.AmazonS3.DownloadObject/DownloadObject.cs

```diff
@@ -0,0 +1,226 @@
+                                results.Add(new SingleResultObject($"File {fileObject.Key.Split('/').Last()} was skipped because it already exists at {fullPath} and ContinueIfExists is true. Set Overwrite to true to overwrite the file."));
+                            else
+                            {
+                                results.Add(new SingleResultObject(await WriteToFile(input, fileObject, client, input.DestinationPath, fullPath)));
+                            
+                                if (input.DeleteSourceFile) await DeleteSourceFile(client, cancellationToken, input.BucketName, fileObject.Key);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (results.Count == 0 && input.ThrowErrorIfNoMatch) throw new Exception($"No matches found with search pattern {input.SearchPattern}");
+            return results;
+        }
+
+        private static async Task<string> WriteToFilePreSigned(string destinationPath, string fileName, Stream responseStream, Input input)
+        {
+            string responseBody;
+            if (!destinationPath.EndsWith(Path.DirectorySeparatorChar.ToString())) destinationPath += Path.DirectorySeparatorChar.ToString();
+            var fullPath = $@"{destinationPath}{fileName}";
+
+            using (var reader = new StreamReader(responseStream)) responseBody = await reader.ReadToEndAsync();
+
+
+            if (File.Exists(fullPath) && !input.Overwrite)
+            {
+                throw new Exception($"File already exists at {fullPath}. Set Overwrite to true to overwrite the file.");
+            }
+            if (File.Exists(fullPath) && input.Overwrite)
+            {
+                var file = new FileInfo(fullPath);
+
+                while (IsFileLocked(file)) Thread.Sleep(1000);
+                File.Delete(fullPath);
+                File.WriteAllText(fullPath, responseBody);
+                return $@"Download complete: {fullPath}";
+            }
+            else
+            {
+                if (!Directory.Exists(destinationPath)) Directory.CreateDirectory(destinationPath);
+                File.WriteAllText(fullPath, responseBody);
+                return $@"Download complete: {fullPath}";
+            }
+        }
+
+        private static async Task<string> WriteToFile(Input input, S3Object fileObject, AmazonS3Client s3Client, string destinationFolder, string fullPath)
+        {
+            string responseBody;
+            var request = new GetObjectRequest
+            {
+                BucketName = input.BucketName,
+                Key = fileObject.Key
+            };
+
+            using (var response = await s3Client.GetObjectAsync(request))
+            using (var responseStream = response.ResponseStream)
+            using (var reader = new StreamReader(responseStream)) responseBody = await reader.ReadToEndAsync();
+            if (File.Exists($@"{fullPath}"))
```

Comment:
You don't need to wrap the path to quotes. You can just pass the path with `File.Exists(fullpath)`.